src/mixture/4C_mixture_remodelfiber-internal.hpp

```diff
@@ -148,22 +188,32 @@ namespace Mixture
       /// @{
       [[nodiscard]] T evaluate_current_homeostatic_fiber_cauchy_stress() const;
       [[nodiscard]] T evaluate_current_fiber_cauchy_stress() const;
-      [[nodiscard]] T evaluate_current_fiber_p_k2_stress() const;
-      [[nodiscard]] T evaluate_d_current_fiber_p_k2_stress_d_lambdafsq() const;
-      [[nodiscard]] T evaluate_d_current_fiber_p_k2_stress_d_lambdar() const;
+      [[nodiscard]] T evaluate_current_fiber_pk2_stress() const;
+      [[nodiscard]] T evaluate_d_current_fiber_pk2_stress_d_lambda_f_sq() const;
+      [[nodiscard]] T evaluate_d_current_fiber_pk2_stress_d_lambda_r() const;
       [[nodiscard]] T
-      evaluate_d_current_growth_evolution_implicit_time_integration_residuum_d_lambdafsq(
+      evaluate_d_current_growth_evolution_implicit_time_integration_residuum_d_lambda_f_sq(
           T dt) const;
       [[nodiscard]] T
-      evaluate_d_current_remodel_evolution_implicit_time_integration_residuum_d_lambdafsq(
+      evaluate_d_current_remodel_evolution_implicit_time_integration_residuum_d_lambda_f_sq(
           T dt) const;
       [[nodiscard]] T evaluate_current_growth_scalar() const;
-      [[nodiscard]] T evaluate_current_lambdar() const;
+      [[nodiscard]] T evaluate_current_lambda_r() const;
+
+      [[nodiscard]] T evaluate_d_current_growth_scalar_d_lambda_f_sq() const;
+      [[nodiscard]] T evaluate_d_current_lambda_r_d_lambda_f_sq() const;
+      [[nodiscard]] T evaluate_d_current_cauchy_stress_d_lambda_f_sq() const;
       /// @}
 
       /// array of G&R states (the last state in the array is the current state)
       std::array<GRState, numstates> states_;
 
+      /// current derivatives w.r.t. lambda_f
+      /// @{
+      T d_growth_scalar_d_lambda_f_sq_ = 0.0;
```

Comment:
I guess the `sq` means squared, right? Is this this the derivative of the `growth_scalar` w.r.t. to the square of `lambda_f` or the square of the derivative of the `growth_scalar` w.r.t. to `lambda_f`?

Reply:
yes, the name is a bit ambiguous. It's w.r.t.the square of lambda_f. I tried to make it clear in the docstring above. I used this term quite a lot in the rest of the file. I didn't added comments everywhere. The exact meaning should get clear from the context.